ckanext/harvest/queue.py

```diff
@@ -167,8 +167,18 @@ def resubmit_objects():
         .filter_by(state='WAITING') \
         .all()
 
+    objects_in_queue = []
+    fetch_routing_key = get_fetch_routing_key()
+
+    for i in range(redis.llen(fetch_routing_key)):
+        item = redis.lpop(fetch_routing_key)
+        objects_in_queue.append(
+            json.loads(item)['harvest_object_id']
+        )
+        redis.rpush(fetch_routing_key, item)
```

Comment:
Instead of popping and re-pushing each item from Redis we can get them all in one go and process them in Python:

```suggestion
    for item in redis.lrange(fetch_routing_key, 0, -1):
        objects_in_queue.append(
            json.loads(item)['harvest_object_id']
        )
```

Or in one line if you are feeling fancy:

```python
objects_in_queue = [json.loads(o)['harvest_object_id'] for o in redis.lrange(fetch_routing_key, 0, -1)]
```

Reply:
Thanks for the improvement! I'll make that change right now. :)